nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/InvokeScriptProcessor.java

```diff
@@ -0,0 +1,862 @@
+                        if (logger.isDebugEnabled()) {
+                            logger.error(message, t);
+                        }
+                    }
+                }
+            }
+
+        } catch (final Throwable t) {
+            final ProcessorLog logger = getLogger();
+            final String message = "Unable to load script: " + t;
+
+            // If the module path has not yet been set, then this script is likely being loaded too early and depends
+            // on modules the processor does not yet know about. If this is the case, it will be reloaded later on
+            // property change (modules) or when scheduled
+            //
+            // Alternatively, if the module is
+            if (modulePath != null) {
+                logger.error(message);
+                if (logger.isDebugEnabled()) {
+                    logger.error(message, t);
+                }
+
+                results.add(new ValidationResult.Builder()
+                        .subject("ScriptValidation")
+                        .valid(false)
+                        .explanation("Unable to load script due to " + t)
+                        .input(scriptPath)
+                        .build());
+            }
+        }
+
+        // store the updated validation results
+        validationResults.set(results);
+
+        // return whether there was any issues loading the configured script
+        return results.isEmpty();
+    }
+
+    /**
+     * Invokes the validate() routine provided by the script, allowing for custom validation code.
+     * This method assumes there is a valid Processor defined in the script and it has been loaded
+     * by the InvokeScriptProcessor processor
+     *
+     * @param context The validation context to be passed into the custom validate method
+     * @return A collection of ValidationResults returned by the custom validate method
+     */
+    @Override
+    protected Collection<ValidationResult> customValidate(final ValidationContext context) {
+        final Processor instance = processor.get();
+        final Collection<ValidationResult> currentValidationResults = validationResults.get();
+
+        // if there was existing validation errors and the processor loaded successfully
+        if (currentValidationResults.isEmpty() && instance != null) {
+            try {
+                // defer to the underlying processor for validation
+                final Collection<ValidationResult> instanceResults = instance.validate(context);
+                if (instanceResults != null && instanceResults.size() > 0) {
+                    // return the validation results from the underlying instance
+                    return instanceResults;
+                }
+            } catch (final Throwable t) {
+                final ProcessorLog logger = getLogger();
+                final String message = "Unable to validate the script Processor: " + t;
+
+                logger.error(message);
```

Comment:
Would just pass the Throwable to the logger here instead of doing it in isDebugEnabled, since the logger will make that check itself - though we should probably catch Exception instead of Throwable?